### ISSUE TITLE
torch.mean() documentation does not mention what type of input it should receive

### ISSUE THREAD

**Issue by onel** [pytorch/pytorch#120173]
### 📚 The doc issue

Should the docs also mention that the input tensor should only have float or complex?

If you try and pass anything else you are encountered with:
`mean(): could not infer output dtype. Input dtype must be either a floating point or complex dtype. Got: Long`

Even though the limitation is not mentioned in the docs

### Suggest a potential alternative/fix

The docs should mention what kind of tensor `.mean()` accepts.

**Comment by rambadi** [comment]
Can you assign to me? I can modify the description as:
"Returns the mean value of all elements in the input tensor. Input dtype must be either a floating point or complex dtype.

**Comment by rambadi** [comment]
We can enhance the description to provide a workaround to the problem as well.
"Returns the mean value of all elements in the input tensor. Input dtype must be either a floating point or complex dtype. For integer dtype you can cast the tensor to float value using the dtype parameter."

**Comment by onel** [comment]
Yup, those sound as good options. I scoured the docs a bit looking at other methods but I couldn't find an example where an additional restriction is set on `Tensor` (just cases like `weight (Tensor, optional) – a manual rescaling ...`) 

We might need to use plain english for this one.

It would be helpful to add this in the top description
`Returns the mean value of all elements in the input tensor. Input dtype must be either a floating point or complex dtype.`

And also in the 2 places for parameters:
`input (Tensor) – the input tensor, must be either a floating point or complex dtype`

**Comment by rambadi** [comment]
The top description change is pretty straight forward. Changing the parameters is a bit more involved as it uses a variable that is used in other functions parameters as well, and may need more thorough review beyond "mean()" itself.  I can commit the top description fairly quickly.

**Comment by rambadi** [comment]
Let me know if this is a better description:
"Returns the mean value of all elements in the input tensor. Input dtype must be either a floating point or complex dtype. For other dtypes consider casting the tensor using the dtype parameter."

**Comment by rambadi** [comment]
Okay, I was able to update the parameter section also. Submitting PR shortly.

**Comment by onel** [comment]
LGTM @rambadi I think you also need to sign the CLA for the PR to be fully compliant.

But we also need for someone from the pytorch org to sign off on the PR.

**Comment by rambadi** [comment]
Yep, checking.
